account_bank_statement_import_mt940_base/models/mt940.py

```diff
@@ -0,0 +1,359 @@
+        subfield = list(filter(lambda a: a != '', subfield))
+        if len(subfield) >= 1 and subfield[0]:
+            transaction.update({'account_number': subfield[0]})
+        if len(subfield) >= 3 and subfield[2]:
+            transaction.update({'partner_name': subfield[2]})
+        return transaction
+
+    def handle_common_subfields(self, transaction, subfields):
+        """Deal with common functionality for tag 86 subfields."""
+        # Get counterpart from CNTP, BENM or ORDP subfields:
+        for counterpart_field in ['CNTP', 'BENM', 'ORDP']:
+            if counterpart_field in subfields:
+                self.get_counterpart(transaction,
+                                     subfields[counterpart_field])
+        if not transaction.get('name'):
+            transaction['name'] = '/'
+        # REMI: Remitter information (text entered by other party on trans.):
+        if 'REMI' in subfields:
+            transaction['name'] += (
+                subfields['REMI'][2]
+                # this might look like
+                # /REMI/USTD//<remittance info>/
+                # or
+                # /REMI/STRD/CUR/<betalingskenmerk>/
+                if len(subfields['REMI']) >= 3 and subfields['REMI'][0] in [
+                    'STRD', 'USTD'
+                ]
+                else
+                '/'.join(x for x in subfields['REMI'] if x)
+            )
+        # EREF: End-to-end reference
+        if 'EREF' in subfields:
+            transaction['name'] += '/'.join(filter(bool, subfields['EREF']))
+        # Get transaction reference subfield (might vary):
+        if transaction.get('ref') in subfields:
+            transaction['ref'] = ''.join(subfields[transaction['ref']])
+
+    def pre_process_data(self, data):
+        matches = []
+        self.is_mt940(line=data)
+        data = data.replace(
+            '-}', '}').replace('}{', '}\r\n{').replace('\r\n', '\n')
+        if data.startswith(':940:'):
+            for statement in data.replace(':940:', '').split(':20:'):
+                match = '{4:\n:20:' + statement + '}'
+                matches.append(match)
+        else:
+            tag_re = re.compile(
+                r'(\{4:[^{}]+\})',
+                re.MULTILINE)
+            matches = tag_re.findall(data)
+        return matches
+
+    def parse(self, data, header_lines=None):
+        """Parse mt940 bank statement file contents."""
+        data = data.decode()
+        data = ''.join([str(x) for x in data if str(x) in printable])
+
+        matches = self.pre_process_data(data)
+        statements = []
+        result = {
+            'currency': None,
+            'account_number': None,
+            'statement': None,
+        }
+        if not header_lines:
+            header_lines = self.get_header_lines()
+        for match in matches:
+            self.is_mt940_statement(line=match)
+            iterator = '\n'.join(
+                match.split('\n')[1:-1]).split('\n').__iter__()
```

Comment:
Should be:

```
            iterator = '\n'.join(
                match.split('\n')[1:]).split('\n').__iter__()
```

instead. Otherwise the last line will be ignored. One of the banks I'm working on uses it so I guess there could be more.